src/LiveComponent/src/EventListener/LiveComponentSubscriber.php

```diff
@@ -344,7 +337,17 @@ private function createResponse(MountedComponent $mounted): Response
 
     private function isLiveComponentRequest(Request $request): bool
     {
-        return $request->attributes->has('_live_component');
+        if (!$request->attributes->has('_live_component')) {
+            return false;
+        }
+
+        if ($this->testMode) {
+            return true;
+        }
+
+        // Except when testing, require the correct content-type in the Accept header.
+        // This also acts as a CSRF protection since this can only be set in accordance with same-origin/CORS policies.
+        return \in_array(self::HTML_CONTENT_TYPE, $request->getAcceptableContentTypes(), true);
```

Comment:
This is the most important part of this PR: we mandate our custom content-type in the `Accept` header.

Reply:
Could we refactor in a way we test this only once per request ?

Reply:
I'm not sure it's worth it: the way to do that would be to use an attribute on the request object, polluting the list of attributes. The check should be fast enough and rare enough.